src/lib/asn1/ASN1Reader.cpp

```diff
@@ -147,12 +147,18 @@ CHIP_ERROR ASN1Reader::GetInteger(int64_t & val)
     ReturnErrorCodeIf(ValueLen > sizeof(int64_t), ASN1_ERROR_VALUE_OVERFLOW);
     ReturnErrorCodeIf(mElemStart + mHeadLen + ValueLen > mContainerEnd, ASN1_ERROR_UNDERRUN);
 
+    // NOLINTBEGIN(clang-analyzer-core.UndefinedBinaryOperatorResult)
+    //
+    // TODO: clang-tidy says that if val is -1, then (val << 8) is not defined.
```

Comment:
@emargolis , could you take a look at this please?

Reply:
Pinged Evgeni on the side, he said he would look into this.

Resolving conversation since existing code already does this so new PR does not make things worse, only better (we now both do not allow new instances of this to happen and added a marker of "this is strange" in the code for this instance).